python/ray/autoscaler/autoscaler.py

```diff
@@ -48,12 +49,6 @@
     # node. This takes precedence over min_workers.
     "max_workers": (int, REQUIRED),
 
-    # The number of workers to launch initially, in addition to the head node.
-    "initial_workers": (int, OPTIONAL),
```

Comment:
I actually find this quite useful

Reply:
What's the use case for it now @richardliaw that wouldn't be handled by a request_cores() call in your notebook / script on the head node? The reason we took it out was that we couldn't come up with any situation where request_cores() wouldn't replace it and figured that having a simpler / one-way-to-do-things design would be less confusing for users.

Reply:
Ah I see; right now, my use case for initial_workers is to start a large cluster, and run an experiment overnight, where in a couple hours the experiment finishes and the cluster should downsize. Importantly, my experiment script was first tested on my local machine and then submitted to a cluster.

I guess I might have misunderstood the functionality of `request_cores` - It'd be great if you could include both docstrings and a section on the documentation for how to use it?

Reply:
Right, the way you'd do this (we are actually doing that) is to, at the start of the script, run `request_cores(redis_address, 2000)` which will ensure the cluster has a minimum of 2000 cores available for you to use.

I'll work on documentation!

Reply:
Actually, after talking to more users, I think keeping `initial_workers` is necessary. It's a workaround that allows us to start GPU workers with a non-GPU head, which is a commonly used feature for training.

Reply:
Ultimately though, I think the non-gpu-head gpu-worker setup should work without the `initial_workers`.

Reply:
Initial workers is a huge hack, because if your nodes go idle, you now need to restart the entire cluster in order to get them to come back up. I think it's way better to request directly.

request_gpus could be added here if necessary.

I've been too busy with other stuff to get documentation written and this upstream but I'd be happy with a request_gpu's option in here.